pkg/fixtures/dackbox.go

```diff
@@ -0,0 +1,1162 @@
+			{
+				Path: "/murderous/cane",
+				Dependencies: []string{
+					"experimental-powder",
+				},
+			},
+		},
+	}
+}
+
+// GetImageSherlockHolmes_1 provides a pseudo-realistic image for dackbox integration testing.
+func GetImageSherlockHolmes_1() *storage.Image {
+	return &storage.Image{
+		Id: "sha256:50fa59cca653c51d194974830826ff7a9d9095175f78caf40d5423d3fb12c4f7",
+		Name: &storage.ImageName{
+			Registry: "baker.st",
+			Remote:   "sherlock/holmes",
+			Tag:      "v1",
+			FullName: "baker.st/sherlock/holmes:v1",
+		},
+		Metadata: &storage.ImageMetadata{
+			V1: &storage.V1Metadata{
+				Digest:  "sha256:0a488a3872bfcd9e79a3575b5c273b01c01a21b16e86213a26eb7f3ab540eb84",
+				Created: &types.Timestamp{Seconds: 1553642092, Nanos: 227945051},
+				Author:  "Sir Arthur Conan Doyle",
+				Layers: []*storage.ImageLayer{
+					{
+						Instruction: "COPY",
+						Value:       "/ / # buildkit",
+						Created:     &types.Timestamp{Seconds: 1553640086, Nanos: 106246179},
+					},
+					{
+						Instruction: " /usr/local/bin/ # buildkit",
+						Value:       "file:4fc310c0cb879c876c5c0f571af765a0d24d36cb9253e0f53a0cda2f7e4c1844 in /",
+						Created:     &types.Timestamp{Seconds: 1553640126, Nanos: 263243615},
+					},
+					{
+						Instruction: "ADD",
+						Value:       "file:4fc310c0cb879c876c5c0f571af765a0d24d36cb9253e0f53a0cda2f7e4c1844 in /",
+						Created:     &types.Timestamp{Seconds: 1553640134, Nanos: 213199897},
+					},
+				},
+				User:       "root",
+				Command:    nil,
+				Entrypoint: nil,
+				Volumes:    nil,
+				Labels:     nil,
+			},
+			V2: &storage.V2Metadata{Digest: "sha256:4d818f38fa9dcbf41e7c255f276a72e5c471c1523b6f755a344bac04652351dd"},
+			LayerShas: []string{
+				"sha256:50fa59cca653c51d194974830826ff7a9d9095175f78caf40d5423d3fb12c4f7",
+				"sha256:e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855",
+				"sha256:8e2ee98ae01ebe81fe221f5a444cab18c8f7a26cd00ce1ef23cf7432feef99b4",
+			},
+			DataSource: &storage.DataSource{
+				Id:   "13e92196-8216-4714-9ac7-fac779bb973b",
+				Name: "Sir Arthur Conan Doyle",
+			},
+			Version: 0,
+		},
+		Scan: &storage.ImageScan{
+			ScannerVersion: "2.24.0-11-g05cf175999",
+			ScanTime:       &types.Timestamp{Seconds: 1654154310, Nanos: 970783800},
+			Components: []*storage.EmbeddedImageScanComponent{
+				GetEmbeddedImageComponent_1_1(),
+				GetEmbeddedImageComponent_1_2(),
+				GetEmbeddedImageComponent_1s2_3(),
+			},
+			OperatingSystem: "crime-stories",
+			DataSource: &storage.DataSource{
+				Id:   "169b0d3f-8277-4900-bbce-1127077defae",
+				Name: "Stackrox Scanner",
+			},
+			Notes: []storage.ImageScan_Note{},
+		},
+		SignatureVerificationData: nil,
+		Signature:                 nil,
+		SetComponents:             &storage.Image_Components{Components: 3},
+		SetCves:                   &storage.Image_Cves{Cves: 5},
+		SetFixable:                &storage.Image_FixableCves{FixableCves: 2},
```

Comment:
Not sure I'm reading things right, but shouldn't it be logically 5 fixable CVE's?

All three embedded image components have the `FixedBy` field set, all three components are different and `1_1 & 1s2_3` have each respectively 2 CVE's and `1_2` has 1 CVE, or are my assumptions wrong that they are not somehow fixed?

Reply:
To be honest, I did not dig that deep in the effort to provide pseudo-realistic test data.
My guess would be that the `SetFixedBy` field in the CVEs from the graph is what is actually used to consider a CVE as fixed. I think the `FixableCves` value is the count of visible CVEs that are flagged as fixable.
Although the values are not relevant in the tests I intend to run, I adjusted the dataset to be consistent.